sonar-auth-saml-plugin/src/main/java/org/sonarsource/auth/saml/SamlIdentityProvider.java

```diff
@@ -19,20 +19,49 @@
  */
 package org.sonarsource.auth.saml;
 
+import com.onelogin.saml2.Auth;
+import com.onelogin.saml2.exception.SettingsException;
+import com.onelogin.saml2.settings.Saml2Settings;
+import com.onelogin.saml2.settings.SettingsBuilder;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 import org.sonar.api.server.ServerSide;
 import org.sonar.api.server.authentication.Display;
 import org.sonar.api.server.authentication.OAuth2IdentityProvider;
+import org.sonar.api.server.authentication.UnauthorizedException;
+import org.sonar.api.server.authentication.UserIdentity;
 import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
 
+import static java.util.Collections.emptySet;
+import static java.util.Objects.requireNonNull;
+
 @ServerSide
 public class SamlIdentityProvider implements OAuth2IdentityProvider {
 
-  static final String KEY = "saml";
+  private static final String KEY = "saml";
 
   private static final Logger LOGGER = Loggers.get(SamlIdentityProvider.class);
 
-  public SamlIdentityProvider() {
+  private final SamlSettings samlSettings;
+  private final boolean strictMode;
+
+  public SamlIdentityProvider(SamlSettings samlSettings) {
+    this(samlSettings, true);
+  }
+
+  SamlIdentityProvider(SamlSettings samlSettings, boolean strictMode) {
```

Comment:
`@VisibleForTesting` isn't missing ?
Because I'm not expecting that strictMode to never be false unless used it tests.